storybookjs/storybook
# Serve correctly hashed static files with the Cache-Control header

Issue: #8278 

By @eashwar and @christopherhafke.

## What we did

After reproducing the behavior described in #8278, we added the `Cache-Control` header to both the static files in the user-specified `staticDir` as well as any files served from `/static/media` (i.e., files loaded and hashed by `file-loader`).

## How to test

To test this change, one can write any story that uses some static file (like an image) located in the `public` or `src` directories of a project. An example of a story that we wrote for `examples/cra-react15`:

```
import logo from '../logo.svg';
...
export const Story3 = () => (
  <Button onClick={action('clicked')}>
    <img src={logo} alt='A nice logo.'/>
  </Button>
);
Story3.story = { name: 'with an image'};
```
Run `yarn storybook`, go to the story (with browser devtools open). The response header for the image file should have the header `Cache-Control`. Reload the page, and the network request made for the image (in this example, `logo.svg`), should say something along the lines of `200 (from memory/disk cache)` in the response header (as opposed to `304 not modified`).

### Is this testable with Jest or Chromatic screenshots?

We're unsure about how we could test something like this with Jest, or if that needs to be done at all - `build-dev.js` and `dev-server.js` don't seem to have tests written for them.

### Does this need a new example in the kitchen sink apps?

This is unlikely, as the PR does not add any new functionality (instead improves existing functionality).

### Does this need an update to the documentation?

No.
